client/allochealth/tracker.go

```diff
@@ -381,8 +455,12 @@ func (t *Tracker) watchConsulEvents() {
 OUTER:
 	for {
 		select {
+
+		// we are shutting down
```

Comment:
This is dangerous phrasing. Is "we" the *agent* shutting down? I believe it just means *this tracker is no longer needed and you don't need to know why.* It could be canceled due to the alloc being stopped or another event causing the health to be set, but I don't think actually shutting down the agent closes it! So maybe:

```suggestion
		// tracker has been canceled, no need to keep waiting
```